module_checksum_upgrade/models/module.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import json
+import logging
+import os
+
+from openerp import api, models
+from openerp.modules.module import get_module_path
+
+_logger = logging.getLogger(__name__)
+try:
+    from checksumdir import dirhash
+except ImportError:
+    _logger.debug('Cannot `import checksumdir`.')
+
+PARAM_INSTALLED_CHECKSUMS = \
+    'module_checksum_upgrade.installed_checksums'
+PARAM_EXCLUDED_EXTENSIONS = \
+    'module_checksum_upgrade.checksum_excluded_extensions'
+
+
+class Module(models.Model):
+    _inherit = 'ir.module.module'
+
+    @api.multi
+    def _get_checksum_dir(self):
+        self.ensure_one()
+
+        excluded_extensions = self.env["ir.config_parameter"].get_param(
+            PARAM_EXCLUDED_EXTENSIONS,
+            'pyc,pyo',
```

Comment:
could we add `.pot` too?

Reply:
Yes, makes sense. Would be nice to have a mechanism to exclude/include languages too.